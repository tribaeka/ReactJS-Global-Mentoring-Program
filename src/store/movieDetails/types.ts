import {IMoviesItem} from "@components/search-results/movies-list/IMoviesItem";

export const SET_MOVIE = 'SET_MOVIE';
export const CLEAR_MOVIE = 'CLEAR_MOVIE';

export interface IMovieDetailsPayload {
    movie: IMoviesItem;
}

interface ISetMovieAction {
    type: typeof SET_MOVIE;
    payload: IMovieDetailsPayload;
}

interface IClearMovieAction {
    type: typeof CLEAR_MOVIE;
}

export type MovieDetailsActionTypes = ISetMovieAction | IClearMovieAction;
