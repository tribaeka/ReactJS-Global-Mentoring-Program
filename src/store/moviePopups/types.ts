import {IMoviesItem} from "../../components/home/main/search-results/movies-list/IMoviesItem";

export const OPEN_MOVIE_POPUP = 'OPEN_MOVIE_POPUP';
export const CLOSE_MOVIE_POPUP = 'CLOSE_MOVIE_POPUP';

export const MOVIE_POPUPS_MAP = {
    ADD: 'addMoviePopup',
    EDIT: 'editMoviePopup',
    DELETE: 'deleteMoviePopup'
}

export interface IMoviePopupsPayload {
    name: string;
    title: string;
    movie?: IMoviesItem;
}

interface IOpenMoviePopupAction {
    type: typeof OPEN_MOVIE_POPUP;
    payload: IMoviePopupsPayload;
}

interface ICloseMoviePopupAction {
    type: typeof CLOSE_MOVIE_POPUP;
}

export type MoviePopupsActionTypes = IOpenMoviePopupAction | ICloseMoviePopupAction