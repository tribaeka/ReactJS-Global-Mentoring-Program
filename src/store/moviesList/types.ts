import {IMoviesItem} from "../../components/home/main/search-results/movies-list/IMoviesItem";

export const GET_MOVIES_LIST = 'GET_MOVIES_LIST';
export const REQUEST_GET_MOVIES_LIST = 'REQUEST_GET_MOVIES_LIST';
export const UPDATE_SORT_BY = 'UPDATE_SORT_BY';
export const UPDATE_FILTER = 'UPDATE_FILTER';


export interface IGetMoviesListPayload {
    data: IMoviesItem[];
    limit: number;
    offset: number;
    totalAmount: number;
}

export interface IUpdateSortByPayload {
    sortBy: string;
}

export interface IUpdateFilterPayload {
    filter: string;
}

interface IGetMoviesListAction {
    type: typeof GET_MOVIES_LIST;
    payload: IGetMoviesListPayload;
}

interface IRequestGetMoviesListAction {
    type: typeof REQUEST_GET_MOVIES_LIST;
}

interface IUpdateSortByAction {
    type: typeof UPDATE_SORT_BY,
    payload: IUpdateSortByPayload
}

interface IUpdateFilterAction {
    type: typeof UPDATE_FILTER,
    payload: IUpdateFilterPayload
}

export type MoviesListActionTypes = IGetMoviesListAction
    | IRequestGetMoviesListAction
    | IUpdateSortByAction
    | IUpdateFilterAction;