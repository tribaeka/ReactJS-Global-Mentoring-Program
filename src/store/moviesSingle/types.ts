import {IMoviesItem} from "../../components/home/main/search-results/movies-list/IMoviesItem";

export const UPDATE_MOVIES_SINGLE = 'UPDATE_MOVIES_SINGLE';
export const REQUEST_UPDATE_MOVIES_SINGLE = 'REQUEST_UPDATE_MOVIES_SINGLE';



interface IUpdateMoviesSingleAction {
    type: typeof UPDATE_MOVIES_SINGLE;
    payload: IMoviesItem;
}

interface IRequestUpdateMoviesSingleAction {
    type: typeof REQUEST_UPDATE_MOVIES_SINGLE;
    payload?: IMoviesItem
}

export type MoviesSingleActionTypes = IUpdateMoviesSingleAction | IRequestUpdateMoviesSingleAction;