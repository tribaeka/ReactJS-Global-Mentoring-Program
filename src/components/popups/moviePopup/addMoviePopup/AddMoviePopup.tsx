import React from 'react';
import CloseBtn, { CloseBtnSizes } from '@components/shared/closeBtn/CloseBtn';
import {connect} from 'react-redux';
import MovieForm from '@components/shared/movieForm/MovieForm';
import { IMoviePopupProps } from '@components/popups/moviePopup/IMoviePopupProps';
import { closePopup } from "@store/moviePopups/actions";
import {compose} from "redux";

const AddMoviePopup: React.FC<IMoviePopupProps> = ({ title, closePopup }) => {

    return (
        <div className="popup movie-popup">
            <div className="movie-popup-close-btn">
                <CloseBtn clickHandler={closePopup} size={CloseBtnSizes.BIG}/>
            </div>
            <h2>{title}</h2>
            <MovieForm method={'POST'}/>
        </div>
    );
}

export default compose(
    connect(null, {closePopup}),
    React.memo
)(AddMoviePopup);
