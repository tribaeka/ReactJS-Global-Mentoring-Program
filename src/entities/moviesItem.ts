import { IMoviesItem, IMoviesServerItem } from "@components/search-results/movies-list/IMoviesItem";

export const MOVIES_ITEM_MOCK: IMoviesItem = {
    budget: 923000,
    genres: "Adventure, Comedy, Drama",
    id: 962,
    overview: "A lone prospector ventures into Alaska looking for gold. He gets mixed up with some burly characters and falls in love with the beautiful Georgia. He tries to win her heart with his singular charm.",
    posterPath: "https://image.tmdb.org/t/p/w500/eQRFo1qwRREYwj47Yoe1PisgOle.jpg",
    releaseDate: "1925-06-25",
    revenue: 2500000,
    runtime: 95,
    tagline: "The World's Greatest Laughing Picture!",
    title: "The Gold Rush",
    voteAverage: 7.8,
    voteCount: 411
};

export const MOVIES_SERVICE_ITEM_MOCK: IMoviesServerItem = {
    budget: 923000,
    genres: ["Adventure", "Comedy", "Drama"],
    id: 962,
    overview: "A lone prospector ventures into Alaska looking for gold. He gets mixed up with some burly characters and falls in love with the beautiful Georgia. He tries to win her heart with his singular charm.",
    poster_path: "https://image.tmdb.org/t/p/w500/eQRFo1qwRREYwj47Yoe1PisgOle.jpg",
    release_date: "1925-06-25",
    revenue: 2500000,
    runtime: 95,
    tagline: "The World's Greatest Laughing Picture!",
    title: "The Gold Rush",
    vote_average: 7.8,
    vote_count: 411
};

export const MOVIES_LIST_MOCK = [
    MOVIES_ITEM_MOCK,
    {
        budget: 92620000,
        genres: "Drama, Science Fiction",
        id: 19,
        overview: "In a futuristic city sharply divided between the working class and the city planners, the son of the city's mastermind falls in love with a working class prophet who predicts the coming of a savior to mediate their differences.",
        posterPath: "https://image.tmdb.org/t/p/w500/qriaeUUwdmlgethK3aSAx68mG05.jpg",
        releaseDate: "1927-01-10",
        revenue: 650422,
        runtime: 153,
        tagline: "There can be no understanding between the hands and the brain unless the heart acts as mediator.",
        title: "Metropolis",
        voteAverage: 8.1,
        voteCount: 837
    }
];
