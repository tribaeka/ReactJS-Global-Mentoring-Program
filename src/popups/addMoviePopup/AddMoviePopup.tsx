import React from 'react';
import CloseBtn, { CloseBtnSizes } from '../../shared/closeBtn/CloseBtn';
import { useDispatch } from 'react-redux';

const AddMoviePopup: React.FC = () => {

    const dispatch = useDispatch();

    function closePopup() {
        dispatch({type: 'CLOSE_ADD_MOVIE_POPUP'})
    }

    return (
        <div className="popup addMoviePopup">
            <div className="add-movie-popup-close-btn">
                <CloseBtn clickHandler={closePopup} size={CloseBtnSizes.BIG}/>
            </div>
            <h1>ADD MOVIE</h1>
            <form className="add-movie-form">
                <div className="form-control">
                    <label className="form-label" htmlFor="">TITLE</label>
                    <input className="form-input" type="text" placeholder="Enter title"/>
                </div>
                <div className="form-control">
                    <label className="form-label" htmlFor="">RELEASE DATE</label>
                    <input className="form-input" type="date" placeholder="Select date"/>
                </div>
                <div className="form-control">
                    <label className="form-label" htmlFor="">MOVIE URL</label>
                    <input className="form-input" type="text" placeholder="Movie URL here"/>
                </div>
                <div className="form-control">
                    <label className="form-label" htmlFor="">GENRE</label>
                    <input className="form-input" type="text" placeholder="Select Genre"/>
                </div>
                <div className="form-control">
                    <label className="form-label" htmlFor="">OVERVIEW</label>
                    <input className="form-input" type="text" placeholder="Overview here"/>
                </div>
                <div className="form-control">
                    <label className="form-label" htmlFor="">RUNTIME</label>
                    <input className="form-input" type="text" placeholder="Runtime here"/>
                </div>
                <div className="form-btn-control">
                    <button className="form-btn" type="reset">RESET</button>
                    <button className="form-btn" type="submit">SUBMIT</button>
                </div>
            </form>
        </div>
    );
}

export default AddMoviePopup;
