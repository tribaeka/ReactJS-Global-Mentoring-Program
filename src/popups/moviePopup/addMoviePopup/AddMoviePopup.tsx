import React from 'react';
import CloseBtn, { CloseBtnSizes } from '../../../shared/closeBtn/CloseBtn';
import { useDispatch } from 'react-redux';
import MovieForm from '../../../shared/movieForm/MovieForm';

const AddMoviePopup: React.FC = () => {

    const dispatch = useDispatch();

    function closePopup() {
        dispatch({type: 'CLOSE_MOVIE_POPUP'})
    }

    return (
        <div className="popup movie-popup">
            <div className="movie-popup-close-btn">
                <CloseBtn clickHandler={closePopup} size={CloseBtnSizes.BIG}/>
            </div>
            <h1>ADD MOVIE</h1>
            <MovieForm/>
        </div>
    );
}

export default AddMoviePopup;
