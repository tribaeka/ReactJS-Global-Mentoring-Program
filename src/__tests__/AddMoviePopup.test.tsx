import { shallow } from "enzyme";
import configureStore from "redux-mock-store";
import AddMoviePopup from "../components/popups/movie-popup/add-movie-popup/AddMoviePopup";
import React from "react";
import { MOVIE_POPUPS_MAP } from "../store/moviePopups/types";
import { MOVIES_ITEM_MOCK } from "../entities/moviesItem";
import { Provider } from "react-redux";

const mockStore = configureStore([]);
const addMoviePopupTitle = 'ADD MOVIE';

describe('it should test Add movie modal dialog', () => {
    let store;
    let component;

    beforeEach(() => {
        store = mockStore({
            isPopupOpened: true,
            popupName: MOVIE_POPUPS_MAP.ADD,
            popupTitle: addMoviePopupTitle,
            movie: MOVIES_ITEM_MOCK
        });
        component = shallow(
            <Provider store={store}>
                <AddMoviePopup title={addMoviePopupTitle}/>
            </Provider>
        )
    });

    it('should render with given state from store', () => {
        expect(component).toMatchSnapshot();
    });

});
