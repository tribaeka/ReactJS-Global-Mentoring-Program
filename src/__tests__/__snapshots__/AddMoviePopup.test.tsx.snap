// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it should test Add movie modal dialog should render with given state from store 1`] = `
<ContextProvider
  value={
    Object {
      "store": Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      },
      "subscription": Subscription {
        "handleChangeWrapper": [Function],
        "listeners": Object {
          "notify": [Function],
        },
        "onStateChange": [Function],
        "parentSub": undefined,
        "store": Object {
          "clearActions": [Function],
          "dispatch": [Function],
          "getActions": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
        },
        "unsubscribe": null,
      },
    }
  }
>
  <Connect(Component)
    title="ADD MOVIE"
  />
</ContextProvider>
`;
